src/accel/AlongStepFactory.cc

```diff
@@ -0,0 +1,89 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/AlongStepFactory.cc
+//---------------------------------------------------------------------------//
+#include "AlongStepFactory.hh"
+
+#include "corecel/io/Logger.hh"
+#include "corecel/math/ArrayUtils.hh"
+#include "celeritas/em/UrbanMscParams.hh"
+#include "celeritas/field/UniformFieldData.hh"
+#include "celeritas/global/alongstep/AlongStepGeneralLinearAction.hh"
+#include "celeritas/global/alongstep/AlongStepUniformMscAction.hh"
+#include "celeritas/io/ImportData.hh"
+
+#include "CLHEP/Units/SystemOfUnits.h"
+#include "detail/Convert.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with a function to return the field strength.
+ *
+ * The function is evaluated whenever Celeritas is set up (which is after
+ * Geant4 physics is initialized).
+ */
+UniformAlongStepFactory::UniformAlongStepFactory(FieldFunction f)
+    : get_field_(std::move(f))
+{
+    CELER_EXPECT(get_field_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Emit an along-step action.
+ *
+ * The action will embed the linear propagator if the magnetic field strength
+ * is zero (or the accessor is unset).
+ */
+auto UniformAlongStepFactory::operator()(AlongStepFactoryInput const& input) const
+    -> result_type
+{
+    // Get the field strength in tesla (or zero if accessor is undefined)
+    Real3 field = get_field_
+                      ? detail::convert_from_geant(get_field_(), CLHEP::tesla)
+                      : Real3{0, 0, 0};
+    real_type magnitude_tesla = norm(field);
+
+    if (magnitude_tesla > 0)
+    {
+        // Create a linear field
+        CELER_VALIDATE(!input.imported->em_params.energy_loss_fluct,
```

Comment:
Is this necessary?

Reply:
To match the AdePT implementation for comparison purposes, we don't include eloss fluctuations. It would be trivial to enable it but would mean instantiating another along-step kernel. I might soon try breaking up along-step into several smaller kernels which would make it easier to mix and match this...

Reply:
Actually I could turn this to a warning: we have the option to warn for missing physics (coulomb scattering) so maybe we should allow this?